source/retryable-writes/tests/unified/insertOne-serverErrors.yml

```
@@ -0,0 +1,78 @@
+description: "retryable-writes insertOne serverErrors"
+
+schemaVersion: "1.0"
+
+runOnRequirements:
+  - minServerVersion: "3.6"
+    topologies: [ replicaset ]
+
+createEntities:
+  - client:
+      id: &client0 client0
+      useMultipleMongoses: false
+      observeEvents: [ commandStartedEvent ]
+  - database:
+      id: &database0 database0
+      client: *client0
+      databaseName: &databaseName retryable-writes-tests
+  - collection:
+      id: &collection0 collection0
+      database: *database0
+      collectionName: &collectionName coll
+
+initialData:
+  - collectionName: *collectionName
+    databaseName: *databaseName
+    documents:
+      - { _id: 1, x: 11 }
+      - { _id: 2, x: 22 }
+
+tests:
+  - description: "InsertOne succeeds after retryable writeConcernError"
+    runOnRequirements:
+      - minServerVersion: "4.0"
+        topologies: [ replicaset ]
+      - minServerVersion: "4.1.7"
+        topologies: [ sharded-replicaset ]
+    operations:
+      - name: failPoint
+        object: testRunner
+        arguments:
+          client: *client0
+          failPoint:
+            configureFailPoint: failCommand
+            mode: { times: 1 }
+            data:
+              failCommands: [ insert ]
+              errorLabels: [RetryableWriteError] # top-level error labels
+              writeConcernError:
+                code: 91 # ShutdownInProgress
+                errmsg: "Replication is being shut down"
+      - name: insertOne
+        object: *collection0
+        arguments:
+          document: { _id: 3, x: 33 }
+        expectResult:
+          insertedId: 3
```

```suggestion
          $$unsetOrMatches: { insertedId: { $$unsetOrMatches: 3 } }
```

See https://github.com/mongodb/specifications/commit/44f83106033a609f2af40e3e4d99852a03096ba5#diff-7b80332d6e4f34d2b705cb9fab2c176444ad002089466b6cf47b3dd24b0b8383R38 for context.

---

That's quite a mouthful.

Done.